src/credentials/sso/ssoAccessTokenProvider.ts

```diff
@@ -0,0 +1,148 @@
+/*!
+ * Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import * as vscode from 'vscode'
+import { ClientRegistration } from './clientRegistration'
+import { AccessToken } from './accessToken'
+import { DiskCache } from './diskCache'
+import { Authorization } from './authorization'
+import { getLogger } from '../../shared/logger'
+
+const CLIENT_REGISTRATION_TYPE = 'public'
+const GRANT_TYPE = 'urn:ietf:params:oauth:grant-type:device_code'
```

Comment:
Mind adding a comment about what this means?

Reply:
Done